dask/diagnostics/profile.py

```diff
@@ -43,10 +43,14 @@ class Profiler(Callback):
     """
     def __init__(self):
         self._results = {}
+        self.results = []
         self._dsk = {}
 
-    def _start(self, dsk):
+    def __enter__(self):
         self.clear()
+        return super(Profiler, self).__enter__()
```

Comment:
This now depends on using the contextmanager syntax, or calling `clear` explicitly after calculations if registered. Probably the only way to accomplish this, but it should be noted.